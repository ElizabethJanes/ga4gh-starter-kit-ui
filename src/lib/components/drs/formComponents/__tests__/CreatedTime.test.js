import React from 'react';
import renderer from 'react-test-renderer';
import CreatedTime from '../CreatedTime';

test('SHOW <CreatedTime /> ', () => {
    const createdTime = renderer
    .create(<CreatedTime 
    readOnly={true}
    created_time='2021-07-22T13:00:00Z' />)
    .toJSON();
    expect(createdTime).toMatchSnapshot();
});

test('NEW and EDIT <CreatedTime />', () => {
    const createdTime = renderer
    .create(<CreatedTime 
    readOnly={false}
    created_time='2021-07-22T13:00:00Z' />)
    .toJSON();
    expect(createdTime).toMatchSnapshot();
});