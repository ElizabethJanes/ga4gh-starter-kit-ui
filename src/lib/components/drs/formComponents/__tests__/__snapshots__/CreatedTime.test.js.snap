// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NEW and EDIT <CreatedTime /> 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-fullWidth"
>
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
    format="yyyy-MM-dd HH:mm:ss OOOO"
    onClick={[Function]}
    onKeyDown={[Function]}
    readOnly={false}
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink={false}
      htmlFor="created_time"
      id="created_time-label"
    >
      Created Time
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-describedby="created_time-helper-text"
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        id="created_time"
        name="created_time"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        readOnly={true}
        required={false}
        type="text"
        value="2021-07-22 09:00:00 GMT-04:00"
      />
    </div>
    <p
      className="MuiFormHelperText-root"
      id="created_time-helper-text"
    >
      Timestamp of DRS Object creation in ISO 8601 format.
    </p>
  </div>
</div>
`;

exports[`SHOW <CreatedTime />  1`] = `
<div
  className="MuiFormControl-root MuiFormControl-fullWidth"
>
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
    format="yyyy-MM-dd HH:mm:ss OOOO"
    onClick={[Function]}
    onKeyDown={[Function]}
    readOnly={true}
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
      data-shrink={true}
      htmlFor="created_time"
      id="created_time-label"
    >
      Created Time
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-describedby="created_time-helper-text"
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        id="created_time"
        name="created_time"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        readOnly={true}
        required={false}
        type="text"
        value="2021-07-22 09:00:00 GMT-04:00"
      />
    </div>
    <p
      className="MuiFormHelperText-root MuiFormHelperText-filled"
      id="created_time-helper-text"
    >
      Timestamp of DRS Object creation in ISO 8601 format.
    </p>
  </div>
</div>
`;
